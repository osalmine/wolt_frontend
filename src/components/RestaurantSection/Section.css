.Section {
	font-size: calc(10px + 2vmin);
	padding-bottom: 0.5rem;
	text-align: left;
	height: 100%;
}

.Section h1 {
	color: #282c34;
	margin-top: 0;
	margin-bottom: 0;
	padding-left: 2rem;
}

.Section:first-of-type {
	margin-top: 7rem;
}

.anchor {
    position: relative;
    top: -85px;
    visibility: hidden;
}

.move-button {
	border: none;
	background: none;
	outline: none;
}

.move-button i {
	font-size: 28px;
	padding: 0 10px;
	/* color: white; */
	color: #282c34;
}

.move-button i:hover {
	color: rgb(156, 156, 156);
}

.control-container {
	/* position: absolute;
	left: 50%;
	transform: translatex(-50%); */
	/* bottom: 10px; */
	top: 90%;
	display: flex;
	justify-content: center;
	align-items: center;
}

.slide-dot {
	border-radius: revert;
}

.slide-height {
	height: 30%;
}