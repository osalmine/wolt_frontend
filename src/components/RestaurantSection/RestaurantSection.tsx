import React from 'react';
import RestaurantComponent from '../Restaurant/Restaurant';
import './Section.css';
import { CarouselProvider, Slider, Slide, ButtonBack, ButtonNext, Dot } from 'pure-react-carousel';
import 'pure-react-carousel/dist/react-carousel.es.css';

export interface Restaurant {
	blurhash: string;
	launch_date: string;
	location: number[];
	name: string;
	online: boolean;
	popularity: number;
	distanceFromCurrentLocation?: number;
}

export interface Section {
	title: string;
	restaurants: Restaurant[];
}

interface Props {
	section: Section;
	displayBlurhash?: boolean;
}

const RestaurantSection: React.FC<Props> = ({ section, displayBlurhash = true }) => {
	return (
    	<div className="Section">
			<i className="anchor" id={section.title.replace(/ /g,'')}/>
			<h1>{section.title}</h1>
			<CarouselProvider
				naturalSlideWidth={160}
				naturalSlideHeight={90}
				totalSlides={section.restaurants.length}
				infinite={true}
			>
				<Slider>
					{section.restaurants.map((restaurant, i) => {
						return (
							<Slide index={i} key={i}>
								<RestaurantComponent restaurant={restaurant} displayBlurhash={displayBlurhash} />
							</Slide>);
					})}
				</Slider>
				<div className="control-container">
					<ButtonBack className="move-button"><i className="fas fa-angle-left fa-3l fa-ccolor" /></ButtonBack>
					{section.restaurants.map((el, i) => <Dot className="slide-dot" slide={i} key={i}/>)}
					<ButtonNext className="move-button"><i className="fas fa-angle-right fa-ccolor" /></ButtonNext>
				</div>
			</CarouselProvider>
		</div>);
}

export default RestaurantSection;