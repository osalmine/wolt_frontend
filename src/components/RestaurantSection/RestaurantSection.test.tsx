import RestaurantSection from './RestaurantSection';
import inputJson from '../../discovery_page.json';
import { cleanup, render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';

afterEach(cleanup);

const sectionInput = inputJson.sections[0];

describe('RestaurantSection component tests', () => {
	test('renders section from sectionInput', () => {
		render(<RestaurantSection section={sectionInput} displayBlurhash={false} />);
	
		const title = screen.getByText(/Popular Restaurants/i);
		expect(title).toBeInTheDocument();
	});

	test('renders a restaurant from sectionInput', () => {
		render(<RestaurantSection section={sectionInput} displayBlurhash={false} />);
		
		const title = screen.getByText('Sea Chain');
		expect(title).toBeInTheDocument();
	});

	test('matches snapshot', () => {
		const tree = renderer.create(<RestaurantSection section={sectionInput} displayBlurhash={false} />).toJSON();
		expect(tree).toMatchSnapshot();
	});
});