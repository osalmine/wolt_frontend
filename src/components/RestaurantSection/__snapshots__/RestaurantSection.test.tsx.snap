// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RestaurantSection component tests matches snapshot 1`] = `
<div
  className="Section"
>
  <i
    className="anchor"
    id="PopularRestaurants"
  />
  <h1>
    Popular Restaurants
  </h1>
  <div
    className="carousel"
  >
    <div
      aria-live="polite"
      className="horizontalSlider___281Ls carousel__slider carousel__slider--horizontal"
      onKeyDown={[Function]}
      role="listbox"
      style={Object {}}
      tabIndex={0}
    >
      <div
        className="carousel__slider-tray-wrapper carousel__slider-tray-wrap--horizontal"
        style={Object {}}
      >
        <ul
          className="sliderTray___-vHFQ sliderAnimation___300FY carousel__slider-tray carousel__slider-tray--horizontal"
          onClickCapture={[Function]}
          onMouseDown={[Function]}
          onTouchCancel={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          style={
            Object {
              "flexDirection": "row",
              "transform": "translateX(0%) translateX(0px)",
              "width": "1000%",
            }
          }
        >
          <li
            aria-selected={true}
            className="slide___3-Nqo slideHorizontal___1NzNV carousel__slide carousel__slide--visible"
            onBlur={[Function]}
            onFocus={[Function]}
            role="option"
            style={
              Object {
                "paddingBottom": "5.625%",
                "width": "10%",
              }
            }
            tabIndex={0}
          >
            <div
              className="slideInner___2mfX9 carousel__inner-slide"
              style={Object {}}
            >
              <div
                className="restaurant-wrapper"
              >
                <div
                  className="text-container"
                >
                  <h2>
                    Sea Chain
                  </h2>
                  <p>
                    online
                     
                    <span
                      className="dot online"
                    />
                  </p>
                </div>
              </div>
            </div>
          </li>
          <li
            aria-selected={false}
            className="slide___3-Nqo slideHorizontal___1NzNV carousel__slide carousel__slide--hidden"
            onBlur={[Function]}
            onFocus={[Function]}
            role="option"
            style={
              Object {
                "paddingBottom": "5.625%",
                "width": "10%",
              }
            }
            tabIndex={-1}
          >
            <div
              className="slideInner___2mfX9 carousel__inner-slide"
              style={Object {}}
            >
              <div
                className="restaurant-wrapper"
              >
                <div
                  className="text-container"
                >
                  <h2>
                    Salt
                  </h2>
                  <p>
                    online
                     
                    <span
                      className="dot online"
                    />
                  </p>
                </div>
              </div>
            </div>
          </li>
          <li
            aria-selected={false}
            className="slide___3-Nqo slideHorizontal___1NzNV carousel__slide carousel__slide--hidden"
            onBlur={[Function]}
            onFocus={[Function]}
            role="option"
            style={
              Object {
                "paddingBottom": "5.625%",
                "width": "10%",
              }
            }
            tabIndex={-1}
          >
            <div
              className="slideInner___2mfX9 carousel__inner-slide"
              style={Object {}}
            >
              <div
                className="restaurant-wrapper"
              >
                <div
                  className="text-container"
                >
                  <h2>
                    Fake Tomato Mafia
                  </h2>
                  <p>
                    online
                     
                    <span
                      className="dot online"
                    />
                  </p>
                </div>
              </div>
            </div>
          </li>
          <li
            aria-selected={false}
            className="slide___3-Nqo slideHorizontal___1NzNV carousel__slide carousel__slide--hidden"
            onBlur={[Function]}
            onFocus={[Function]}
            role="option"
            style={
              Object {
                "paddingBottom": "5.625%",
                "width": "10%",
              }
            }
            tabIndex={-1}
          >
            <div
              className="slideInner___2mfX9 carousel__inner-slide"
              style={Object {}}
            >
              <div
                className="restaurant-wrapper"
              >
                <div
                  className="text-container"
                >
                  <h2>
                    Charming Pepper Emporium
                  </h2>
                  <p>
                    online
                     
                    <span
                      className="dot online"
                    />
                  </p>
                </div>
              </div>
            </div>
          </li>
          <li
            aria-selected={false}
            className="slide___3-Nqo slideHorizontal___1NzNV carousel__slide carousel__slide--hidden"
            onBlur={[Function]}
            onFocus={[Function]}
            role="option"
            style={
              Object {
                "paddingBottom": "5.625%",
                "width": "10%",
              }
            }
            tabIndex={-1}
          >
            <div
              className="slideInner___2mfX9 carousel__inner-slide"
              style={Object {}}
            >
              <div
                className="restaurant-wrapper"
              >
                <div
                  className="text-container"
                >
                  <h2>
                    Loving Meat Basket
                  </h2>
                  <p>
                    online
                     
                    <span
                      className="dot online"
                    />
                  </p>
                </div>
              </div>
            </div>
          </li>
          <li
            aria-selected={false}
            className="slide___3-Nqo slideHorizontal___1NzNV carousel__slide carousel__slide--hidden"
            onBlur={[Function]}
            onFocus={[Function]}
            role="option"
            style={
              Object {
                "paddingBottom": "5.625%",
                "width": "10%",
              }
            }
            tabIndex={-1}
          >
            <div
              className="slideInner___2mfX9 carousel__inner-slide"
              style={Object {}}
            >
              <div
                className="restaurant-wrapper"
              >
                <div
                  className="text-container"
                >
                  <h2>
                    Vegetable Factory
                  </h2>
                  <p>
                    online
                     
                    <span
                      className="dot online"
                    />
                  </p>
                </div>
              </div>
            </div>
          </li>
          <li
            aria-selected={false}
            className="slide___3-Nqo slideHorizontal___1NzNV carousel__slide carousel__slide--hidden"
            onBlur={[Function]}
            onFocus={[Function]}
            role="option"
            style={
              Object {
                "paddingBottom": "5.625%",
                "width": "10%",
              }
            }
            tabIndex={-1}
          >
            <div
              className="slideInner___2mfX9 carousel__inner-slide"
              style={Object {}}
            >
              <div
                className="restaurant-wrapper"
              >
                <div
                  className="text-container"
                >
                  <h2>
                    Cucumber Hotel
                  </h2>
                  <p>
                    online
                     
                    <span
                      className="dot online"
                    />
                  </p>
                </div>
              </div>
            </div>
          </li>
          <li
            aria-selected={false}
            className="slide___3-Nqo slideHorizontal___1NzNV carousel__slide carousel__slide--hidden"
            onBlur={[Function]}
            onFocus={[Function]}
            role="option"
            style={
              Object {
                "paddingBottom": "5.625%",
                "width": "10%",
              }
            }
            tabIndex={-1}
          >
            <div
              className="slideInner___2mfX9 carousel__inner-slide"
              style={Object {}}
            >
              <div
                className="restaurant-wrapper"
              >
                <div
                  className="text-container"
                >
                  <h2>
                    Horrific Lettuce
                  </h2>
                  <p>
                    online
                     
                    <span
                      className="dot online"
                    />
                  </p>
                </div>
              </div>
            </div>
          </li>
          <li
            aria-selected={false}
            className="slide___3-Nqo slideHorizontal___1NzNV carousel__slide carousel__slide--hidden"
            onBlur={[Function]}
            onFocus={[Function]}
            role="option"
            style={
              Object {
                "paddingBottom": "5.625%",
                "width": "10%",
              }
            }
            tabIndex={-1}
          >
            <div
              className="slideInner___2mfX9 carousel__inner-slide"
              style={Object {}}
            >
              <div
                className="restaurant-wrapper"
              >
                <div
                  className="text-container"
                >
                  <h2>
                    Real Pizza Factory
                  </h2>
                  <p>
                    online
                     
                    <span
                      className="dot online"
                    />
                  </p>
                </div>
              </div>
            </div>
          </li>
          <li
            aria-selected={false}
            className="slide___3-Nqo slideHorizontal___1NzNV carousel__slide carousel__slide--hidden"
            onBlur={[Function]}
            onFocus={[Function]}
            role="option"
            style={
              Object {
                "paddingBottom": "5.625%",
                "width": "10%",
              }
            }
            tabIndex={-1}
          >
            <div
              className="slideInner___2mfX9 carousel__inner-slide"
              style={Object {}}
            >
              <div
                className="restaurant-wrapper"
              >
                <div
                  className="text-container"
                >
                  <h2>
                    Rosemary
                  </h2>
                  <p>
                    offline
                     
                    <span
                      className="dot offline"
                    />
                  </p>
                </div>
              </div>
            </div>
          </li>
        </ul>
      </div>
    </div>
    <div
      className="control-container"
    >
      <button
        aria-label="previous"
        className="buttonBack___1mlaL carousel__back-button move-button"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        <i
          className="fas fa-angle-left fa-3l fa-ccolor"
        />
      </button>
      <button
        aria-label="slide dot"
        className="dot___3c3SI carousel__dot carousel__dot--0 carousel__dot--selected slide-dot"
        disabled={true}
        onClick={[Function]}
        type="button"
      />
      <button
        aria-label="slide dot"
        className="dot___3c3SI carousel__dot carousel__dot--1 slide-dot"
        disabled={false}
        onClick={[Function]}
        type="button"
      />
      <button
        aria-label="slide dot"
        className="dot___3c3SI carousel__dot carousel__dot--2 slide-dot"
        disabled={false}
        onClick={[Function]}
        type="button"
      />
      <button
        aria-label="slide dot"
        className="dot___3c3SI carousel__dot carousel__dot--3 slide-dot"
        disabled={false}
        onClick={[Function]}
        type="button"
      />
      <button
        aria-label="slide dot"
        className="dot___3c3SI carousel__dot carousel__dot--4 slide-dot"
        disabled={false}
        onClick={[Function]}
        type="button"
      />
      <button
        aria-label="slide dot"
        className="dot___3c3SI carousel__dot carousel__dot--5 slide-dot"
        disabled={false}
        onClick={[Function]}
        type="button"
      />
      <button
        aria-label="slide dot"
        className="dot___3c3SI carousel__dot carousel__dot--6 slide-dot"
        disabled={false}
        onClick={[Function]}
        type="button"
      />
      <button
        aria-label="slide dot"
        className="dot___3c3SI carousel__dot carousel__dot--7 slide-dot"
        disabled={false}
        onClick={[Function]}
        type="button"
      />
      <button
        aria-label="slide dot"
        className="dot___3c3SI carousel__dot carousel__dot--8 slide-dot"
        disabled={false}
        onClick={[Function]}
        type="button"
      />
      <button
        aria-label="slide dot"
        className="dot___3c3SI carousel__dot carousel__dot--9 slide-dot"
        disabled={false}
        onClick={[Function]}
        type="button"
      />
      <button
        aria-label="next"
        className="buttonNext___2mOCa carousel__next-button move-button"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        <i
          className="fas fa-angle-right fa-ccolor"
        />
      </button>
    </div>
  </div>
</div>
`;
